compiler/haskell-ide-core/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -184,3 +174,155 @@ prettyFileDiagnostics (filePath, diags) =
 getDiagnosticsFromStore :: StoreItem -> [Diagnostic]
 getDiagnosticsFromStore (StoreItem _ diags) =
     toList =<< Map.elems diags
+
+-- | This represents every diagnostic in a LSP project, the stage type variable is
+--   the type of the compiler stages, in this project that is always the Key data
+--   type found in Development.IDE.State.Shake
+newtype ProjectDiagnostics stage = ProjectDiagnostics {getStore :: DiagnosticStore}
+    deriving Show
+
+prettyDiagnostics :: ProjectDiagnostics stage -> Doc SyntaxClass
+prettyDiagnostics ds =
+    slabel_ "Compiler errors in" $ vcat $ concatMap fileErrors storeContents where
+
+    fileErrors :: (FilePath, [(T.Text, [LSP.Diagnostic])]) -> [Doc SyntaxClass]
+    fileErrors (filePath, stages) =
+        [ slabel_ "File:" $ pretty filePath
+        , slabel_ "Errors:" $ vcat $ map prettyStage stages
+        ]
+
+    prettyStage :: (T.Text, [LSP.Diagnostic]) -> Doc SyntaxClass
+    prettyStage (stage,diags) =
+        slabel_ ("Stage: "<>T.unpack stage) $ vcat $ concatMap diagSC diags
```

Comment:
Let's not include the stage labels. They've never been helpful.

Reply:
They are pretty useful for debugging, they tell you which shake rule the diagnostics originate from. That being said, they're probably not so useful for the user